### PR TITLE
depth peel example

### PR DESCRIPTION
**Pull request by pailhead**
Hi, I've seen requests for something like this pop up in forums and on stack overflow. This is my naive implementation of depth peeling. I need to do more research on this myself but this shows the basic idea of the approach. I was wondering if it could be a useful example?

https://raw.githack.com/pailhead/three.js/depth-peel-example/examples/webgl_materials_depthpeel.html

I'm also not entirely sure if i even handled the blending correctly, since it is a bit different than the default behavior, but i can't tell is it because it's missing pieces below whats being rendered.

### CODE REVIEW

**Comment by mrdoob** [comment]
Looking good!

I would rename `defaultMode` to `enabled`. And... how's monochrome related to this feature?

File: examples/files.js [hunk 1]
```diff
@@ -275,6 +275,7 @@ var files = {
 		"webgl_water_flowmap"
 	],
 	"webgl / advanced": [
+		"materials_depth_peel",
```
Comment (mrdoob) [inline examples/files.js]: `webgl_materials_depthpeel`

**Comment by pailhead** [comment]
monochromatic was just left there from debugging, to see the effect different. I was thinking of adding a more complex mesh (i struggled with transparency on mechanical parts, like plates with holes) but is it needed?

**Comment by pailhead** [comment]
Actually it's fairly self explanatory. All i can think of from my naive approach is trying to write to the main buffer instead of an off screen one. I'll remove the [wip].

**Comment by pailhead** [comment]
This one's really messy but it does it with less buffers, and stencil i hope might speed things up a bit. I'm struggling getting the same blending though:

https://raw.githack.com/pailhead/three.js/depth-peel-stencil/examples/webgl_materials_depthpeel.html

**Comment by pailhead** [comment]
Any concerns on landing this?

**Comment by stur86** [comment]
I'm following this too. Where is the best place to get the most up-to-date code right now, so I can try and see if I can make my own implementation?

**Comment by pailhead** [comment]
Ah yes. It’s all contained within that example file. Unfortunately the other approach doesn’t work, because threejs locks the render buffers to render targets :(

**Comment by DusanOpenSpace** [comment]
@mrdoob 

I need to remind myself of the state of this example, i think the subsequent one was much faster and robust, however, it depends on:
https://github.com/mrdoob/three.js/pull/15490

I'm not sure rescheduling this does it justice, as it may be far from something that can be actually used in production. It also does not have the depth precision fix that was uncovered recently.

**Comment by Mugen87** [comment]
Closing in favor of #24903.

**Comment by trusktr** [comment]
This one has significantly better visual results than #24903 in depending on the use case. It would be nice if multiple OIT methods could be accepted, with differing API names, rather than only a single one.

Having this as just an example would be great, until someone else can take it further to integrate it as a feature in a renderer and/or material.

**Pull request closed without merging**